### PR TITLE
Adds OpenProps + PostCSS template

### PR DESCRIPTION
**Pull request by joshnuss**
- Adds a template for OpenProps
- Adds icon for OpenProps
- Removed `autoprefixer` package and replaced `stage` in `postcss-preset-env` config with `minimumVendorImplementations` in PostCSS config
  Based on suggestion here: https://twitter.com/romainmenke/status/1682516646462648320 

## Screenshot

image

### CODE REVIEW

**Comment by joshnuss** [comment]
**Note**: Casing of template anem is a bit off

 `Openprops` should be `OpenProps`, and `Postcss` should be `PostCSS`.
 
 But I think it's a different problem, ie. templates should have a way to specify it's label.

**Comment by joshnuss** [comment]
Should we call the template "OpenProps" or "OpenProps + PostCSS"?

I see other templates are using PostCSS too, so maybe specifying that isn't necessary?

**Comment by SarcevicAntonio** [comment]
I'll try my best to take a look ASAP.
Probably tomorrow evening, latest Monday evening GMT+2 ✌️

**Comment by SarcevicAntonio** [comment]
OH and Thanks so much for the PR 😁

**Comment by joshnuss** [comment]
Most welcome!

And no rush, enjoy the weekend 🌞

File: src/lib/default_project_files/openprops/package-lock.json
Comment (paoloricciuti) [inline src/lib/default_project_files/openprops/package-lock.json:1]: We usually delete the package-lock to avoid sending too much over the wire.

**Comment by paoloricciuti** [comment]
Other than the only comment I'd say it's ok...as per the issue of the naming I was thinking: could you create a map of "overrides" and look there leaving the fallback to the folder name transformation? I did this because I wanted to avoid having two places to modify when adding a new template but if we have a fallback should be less of a hassle

**Comment by paoloricciuti** [comment]
Ah also would you consider handling dark mode in global.css (it just need a media prefers, don't worry about the saved preference of the user)

**Comment by joshnuss** [comment]
Thanks @paoloricciuti!

- Removed the `package-lock.json`
- Added dark mode styling. Let me know if that looks OK, (I didn't bring over the `--sk-` vars from the basic template, figured OpenProps provides a similar thing already)

**Comment by paoloricciuti** [comment]
> Thanks @paoloricciuti!
> 
> * Removed the `package-lock.json`
> * Added dark mode styling. Let me know if that looks OK, (I didn't bring over the `--sk-` vars from the basic template, figured OpenProps provides a similar thing already)

Thanks to you for contributing :pray:

LGTM!

**Comment by joshnuss** [comment]
> per the issue of the naming..

Could also be solved with translations. Like if SvelteLab gets translated into other languages, then the translation for "openprops" could correct the case.

**Comment by paoloricciuti** [comment]
> > per the issue of the naming..
> 
> Could also be solved with translations. Like if SvelteLab gets translated into other languages, then the translation for "openprops" could correct the case.

I don't think we plan to translate sveltelab...so I'll probably fix this with the overrides as soon as I have time. Thanks again for contributing 🎉

**Comment by joshnuss** [comment]
Welcome!

> I don't think we plan to translate sveltelab...so I'll probably fix this with the overrides as soon as I have time.
Sounds good 👍🏻

**Pull request merged**